### ISSUE TITLE
Automatic byte count for spectral read

### ISSUE THREAD

**Issue by samiebee43** [WasatchPhotonics/TinyWP#8]
It would make sense to know the correct number of bytes to read during spectral acquisition using the get line length function. 

We do not want to call that from within get_spectrum for two reasons. (1) It's against one of the design principles of this module to have driver functions call each other (2) when get_spectrum is called repeatedly that would correspond to reading the EEPROM repeatedly, which may be slow or reduce the lifespan of the EEPROM (not sure).

https://github.com/WasatchPhotonics/TinyWP/blob/03b0ecd22e89e7a954bc233d743667b10446252e/TinyWP/TinyWP.py#L129-L130

https://github.com/WasatchPhotonics/TinyWP/blob/03b0ecd22e89e7a954bc233d743667b10446252e/TinyWP/TinyWP.py#L117-L124

One option which fits into the rationale of this module is to leave that as a responsibility of the caller. So they would write a client program something along the lines of the following. Here it is the caller who keeps track of the EEPROM data and makes the decision to call it only once at the beginning. That keeps the module from being stateful. We can simplify the caller's work slightly be adding a `get_line_bytes` function which includes the factor of two.

```python3
# ...
bytes_to_read = 2*get_line_length(device) # OR bytes_to_read = get_line_bytes(device)

while True:
    get_spectrum(device, bytes_to_read)
    # ...
```

It's also possible to [use caching](https://stackoverflow.com/questions/815110/is-there-a-decorator-to-simply-cache-function-return-values) on eeprom reading operations so then statefulness is tucked into the language. With caching in place, it could be acceptable to call `get_line_length` from within `get_spectrum`.

In summary the options are:
1. Responsibility of caller: add a `get_line_bytes` function to simplify the client program determining and storing that data
2. Automatic Caching: let `get_spectrum` be an exception to the rule that driver functions do not call each other. Use an annotation for caching
3. Manual Caching: implement the contents of `get_line_length` within get_spectrum (keeping the functions independent). Use a small amount of state to cache the byte_count value so that the USB command that determines the line length only need to be done on the first call of `get_spectrum`
